File: src/python/PythonSDK/foundationallm/models/metadata/knowledge_management_agent.py
"""
Class Name: KnowledgeManagementAgent

Description:
    Encapsulates the metadata for the agent
    fulfilling the orchestration request.
"""
from typing import Optional, List
from pydantic import validator
from .agent_base import AgentBase

class KnowledgeManagementAgent(AgentBase):
    """Knowlege Management Agent metadata model."""
    indexing_profile_object_ids: Optional[List[str]] = None
    text_embedding_profile_object_id: Optional[str] = None

File: src/dotnet/Common/Models/Agents/KnowledgeManagementAgent.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Agents
{
    /// <summary>
    /// The Knowledge Management agent metadata model.
    /// </summary>
    public class KnowledgeManagementAgent : AgentBase
    {
        /// <summary>
        /// The vectorization content source profile.
        /// </summary>
        [JsonPropertyName("content_source_profile_object_id")]
        public string? ContentSourceProfileObjectId { get; set; }

        /// <summary>
        /// The vectorization indexing profile resource path.
        /// </summary>
        [JsonPropertyName("indexing_profile_object_ids")]
        public List<string>? IndexingProfileObjectIds { get; set; }

        /// <summary>
        /// The vectorization text embedding profile resource path.
        /// </summary>
        [JsonPropertyName("text_embedding_profile_object_id")]
        public string? TextEmbeddingProfileObjectId { get; set; }

        /// <summary>
        /// The vectorization text partitioning profile resource path. 
        /// </summary>
        [JsonPropertyName("text_partitioning_profile_object_id")]
        public string? TextPartitioningProfileObjectId { get; set; }

        /// <summary>
        /// Set default property values.
        /// </summary>
        public KnowledgeManagementAgent() =>
            Type = AgentTypes.KnowledgeManagement;
    }
}
